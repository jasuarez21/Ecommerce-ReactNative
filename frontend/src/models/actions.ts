import { Moto } from "./moto";
import { User } from "./user";

export type MotoActions = 
    | {type: 'GET_MOTOS'; data: Array<Moto>}
    | {type: 'ADD_TO_CART'; moto: Moto}
    | {type: 'LOAD_CART';}
    | {type: 'DELETE_CART'; moto: Moto}
    | {type: 'LOG_IN'; data: User}
    | {type: 'LOG_OUT'; data: User}
    | {type: 'SIGN_UP'; data: User}
    | {type: 'UPDATE_CART'}
    | {type: 'DELETE_ACCOUNT'}
    | {type: 'ERROR'}