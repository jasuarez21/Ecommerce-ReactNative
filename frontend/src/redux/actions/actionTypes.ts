const actionTypes = {
  GET_MOTOS: 'GET_MOTOS',
  LOG_IN: 'LOG_IN',
  LOG_OUT: 'LOG_OUT',
  SIGN_UP: 'SIGN_UP',
  ADD_TO_CART: 'ADD_TO_CART',
  LOAD_CART: 'LOAD_CART',
  DELETE_CART: 'DELETE_CART',
  UPDATE_CART: 'UPDATE_CART',
  SAVE_PRODUCT: 'SAVE_PRODUCT',
  DELETE_ACCOUNT: 'DELETE_ACCOUNT',
  ERROR: 'ERROR'
};

export default actionTypes;
