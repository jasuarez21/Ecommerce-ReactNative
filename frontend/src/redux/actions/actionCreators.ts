import { User } from './../../models/user';
import axios from 'axios';
import {URL_MOTOS, URL_LOGIN, URL_LOGOUT, URL_SIGNUP, URL_DELETE_ACCOUNT, URL_SAVE_PRODUCTS} from '@env';
import actionTypes from './actionTypes';
import {Moto} from '../../models/moto';
import {Dispatch} from 'redux';
import {MotoActions} from '../../models/actions';
import { Order } from '../../models/order';

export function GetMotos() {
  return async (dispatch: Dispatch) => {
      try {   
      const {data} = await axios.get(URL_MOTOS);
      dispatch({
        type: actionTypes.GET_MOTOS,
        data,
      });
    } catch (error) {
      dispatch({
        type: actionTypes.ERROR,
      });
    }
  }
}



export function login(email: string, password: string) {
  return async (dispatch: Dispatch) => {
      try {   
        const {data} = await axios.post(URL_LOGIN, {email, password});
        dispatch({
          type: actionTypes.LOG_IN,
          data,
        });
    } catch (error) {
      dispatch({
        type: actionTypes.ERROR,
      });
    }
  }
}

export function logout() {
  return async (dispatch: Dispatch) => {
      try {   
        const {data} = await axios.post(URL_LOGOUT);
        dispatch({
          type: actionTypes.LOG_OUT,
          data,
        });
    } catch (error) {
      dispatch({
        type: actionTypes.ERROR,
      });
    }
  }
}


export function signUp(email: string, password: string, alias: string, avatar: string) {
  return async (dispatch: Dispatch) => {
      try {   
        const {data} = await axios.post(URL_SIGNUP, {email, password, alias, avatar});
        dispatch({
          type: actionTypes.SIGN_UP,
          data,
        });
    } catch (error) {
      dispatch({
        type: actionTypes.ERROR,
      });
    }
  }
}

export function addToCart(moto: Moto): MotoActions {
  return {
    type: 'ADD_TO_CART',
    moto
  };
}

export function loadCart(): MotoActions {
  return {
    type: 'LOAD_CART'
  };
}

export function deleteFromCart(moto: Moto): MotoActions {
  return {
    type: 'DELETE_CART',
    moto,
  };
}

export function reloadCart(moto: Array<Moto>){
  return async (dispatch: Dispatch) => {
    try {   
      for(let i = 0; i < moto.length; i++){
        await axios.post(`${URL_MOTOS}/${moto[i]._id}`, moto[i]);
        dispatch({
          type: actionTypes.UPDATE_CART
        });
      }
  } catch (error) {
    dispatch({
      type: actionTypes.ERROR,
    });
  }
}
}

export function saveOrder(user: User, cart: Order, token: string){
  user.cart.push(cart);
  const config = { 
    headers: {
        authorization: `Bearer ${token}`,
    }
  }
    return async (dispatch: Dispatch) => {
      try {   
          await axios.post(URL_SAVE_PRODUCTS, {_id: user._id, cart: user.cart}, config);
          dispatch({
            type: actionTypes.SAVE_PRODUCT
          });
    } catch (error) {
      dispatch({
        type: actionTypes.ERROR,
      });
    }
  }
}

export function deleteAccount(user: User, token: string){
    return async (dispatch: Dispatch) => {
      try {   
          await axios.delete(URL_DELETE_ACCOUNT, {headers: {
              authorization: `Bearer ${token}`,
          },
              data:{_id: user._id}
          });
          dispatch({
            type: actionTypes.DELETE_ACCOUNT
          });
    } catch (error) {
      dispatch({
        type: actionTypes.ERROR,
      });
    }
  }
}