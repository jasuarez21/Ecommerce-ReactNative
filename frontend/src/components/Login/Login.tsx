import React, {useState} from 'react';
import { Avatar } from 'react-native-elements';
import {Text, TextInput, TouchableHighlight, ImageBackground, Alert, Image, StyleSheet} from 'react-native';
import {StackScreenProps} from '@react-navigation/stack';
import {logout, deleteAccount, login} from '../../redux/actions/actionCreators'
import {useSelector, useDispatch} from 'react-redux';
import {RootState} from '../../redux/store/index'

interface Props extends StackScreenProps<any, any> {}

const Login = ({navigation}: Props) => {
  const user = useSelector((store:RootState) => store.user);
  const dispatch = useDispatch();
  let [email, setEmail] = useState('');
  let [password, setPassword] = useState('');
  function loginClick() {
    dispatch(login(email, password));
  }
  return (
    <>
  <ImageBackground
      source={{uri: 'https://i.postimg.cc/htkHJcp0/fondo-INICIAL.jpg'}}
      style={styles.background}>
        <Image
        style={styles.logo}
        source={{
          uri: 'https://i.postimg.cc/FF2Y5GJd/logo-small.png',
        }}
      />
    {user.token === undefined ?
    (
      <>
      <Text style={styles.textinfo}>Rellena los campos para loguearte en nuestra pagina</Text>
    <>
      <TextInput
        style={styles.input}
        placeholder="email"
        autoCapitalize="none"
        onChangeText={emailInput => setEmail(emailInput)}
        defaultValue={email}
      />
      <TextInput
       style={styles.input}
        placeholder="password"
        secureTextEntry={true}
        autoCapitalize="none"
        onChangeText={passwordInput => setPassword(passwordInput)}
        defaultValue={password}
      />
      <TouchableHighlight onPress={() => loginClick()}>
        <Image
        testID= 'login' 
        source={{uri: "https://i.postimg.cc/6qnQSnBG/icons8-ok-26-1.png"}}
        style={styles.iconOK}
        />
      </TouchableHighlight>
      </>
    <>
      <Text style={styles.textinfo}>Todav√≠a no estas registrado?</Text>
      <TouchableHighlight style={styles.buttonCart} onPress={() => navigation.navigate('SignIn')}>
        <Image 
        testID= 'signup'
        source= {{ uri: 'https://i.postimg.cc/0jfF6Sbg/icons8-add-user-male-64.png'}}
        style={styles.icon}
        />
      </TouchableHighlight>
      </>
    </>
    ) : (
      <>
      {navigation.navigate('Lobby')}
        <TouchableHighlight
          onPress={() => dispatch(logout())}>
          <Text style={styles.logout}
            testID = 'logoutText'
          >SALIR</Text>
        </TouchableHighlight>
        <Avatar
            size="large"
            containerStyle={{ marginLeft: 180}}
            rounded
            source={{
                uri:user?.user?.avatar,
            }}
          />
        <Text style={styles.textprofileTitle}>PERFIL</Text>
        <Text style={styles.textprofile}>Alias: {user?.user?.alias}</Text>
        <Text style={styles.textprofile}>Email: {user?.user?.email}</Text>
        <TouchableHighlight
          onPress={() => Alert.alert(
            "CUIDADO!",
            `Seguro que quieres borrar ${user.user.alias}?`,
            [
              { text: "BORRAR", onPress: () => (
                dispatch(deleteAccount(user.user, user.token)),
                navigation.navigate('Lobby')
                )}
            ]
          )}>
          <Text style={styles.deleteAccount}
            testID = 'deleteAccountButton'
          >Borrar Cuenta</Text>
        </TouchableHighlight>
      </>
    )
    }
    </ImageBackground>
</>
  );
};

const styles = StyleSheet.create({
  background: {
    width: 500,
    height: 800
  },
  textprofile: {
    color: 'white',
    fontSize: 18,
    marginTop: 30,
    marginLeft: 70
  },
  deleteAccount: {
    marginTop: 40,
    marginLeft: 70,
    width: 90,
    textAlign: 'center',
    color: 'black',
    backgroundColor: '#FFB800',
    borderRadius: 5
  },
  textprofileTitle: {
    color: 'white',
    fontSize: 18,
    marginTop: 30,
    marginRight: 70,
    alignSelf: 'center',
    borderBottomWidth: 1,
    borderBottomColor: '#FFB800',
    width: 300
  },
  logout: {
    marginLeft: 340,
    marginTop: 20,
    width: 60,
    textAlign: 'center',
    color: 'black',
    backgroundColor: '#FFB800',
    borderRadius: 5
  },
  iconOK:{
    width: 35,
    height: 35,
    alignSelf: 'center',
    marginTop: 5 ,
    marginRight: 90
  },
  icon: {
    width: 30,
    height: 30,
    alignSelf: 'center',
    marginTop: 1 
  },
  logo: {
    width: 300,
    height: 42,
    marginTop: 80,
    marginBottom: 30,
    marginLeft: 50
  },
  input: {
    backgroundColor: 'white',
    alignSelf: 'center',
    marginBottom: 10,
    marginRight: 70,
    paddingLeft: 20,
    borderRadius: 5,
    width: 300
  },
  buttonCart: {
    backgroundColor: '#FFB800',
    width: 40,
    height: 35,
    display: 'flex',
    alignContent: 'center',
    borderRadius: 5,
    marginLeft: 185,
  },
  textinfo: {
    marginTop: 50,
    marginRight: 90,
    marginBottom: 30,
    fontSize: 16,
    color: 'white',
    alignSelf: 'center'
  },
});


export default Login;
