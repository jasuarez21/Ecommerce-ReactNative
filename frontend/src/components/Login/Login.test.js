import { render, fireEvent } from '@testing-library/react-native';
import React from 'react';
import ConfigureStore from 'redux-mock-store';
import * as actions from '../../redux/actions/actionCreators';
import { Provider } from 'react-redux';
import Login from './Login';

jest.mock('../../redux/actions/actionCreators');

describe('When invoked a Login component', () => {
    const mockStore = ConfigureStore();
    const store = mockStore({
        user: [{
              _id: '2',
              user : {
                  alias: 'Anna',
                  email: 'annita',
                  password: 'annaaaza',
                  avatar: 'img.com'
              }
          }
        ]
    });
    let navigation;
    beforeEach(() => {
        jest.spyOn(actions, 'login').mockReturnValueOnce({ type: '' });
        navigation = {
            navigate: jest.fn()
        };
      });

    test('Should render the login component with inputs', () => {
        const login = render(<Provider store={store}><Login navigation={navigation} /></Provider>)
        expect(login).toMatchSnapshot();
    })
    test('Then setEmail is invoked', () => {
        const { getByPlaceholderText } = render(<Provider store={store}><Login navigation={navigation}/></Provider>);
        const image = getByPlaceholderText('email');
        let setEmail = jest.fn();
        fireEvent.changeText(image, setEmail);
        expect(setEmail).toHaveBeenCalled();
      });
      test('Then setPassword is invoked', () => {
        const { getByPlaceholderText } = render(<Provider store={store}><Login navigation={navigation}/></Provider>);
        const image = getByPlaceholderText('password');
        let setPassword = jest.fn();
        fireEvent.changeText(image, setPassword);
        expect(setPassword).toHaveBeenCalled();
      });
      test('Then login function is invoked', () => {
        const { getByTestId } = render(<Provider store={store}><Login navigation={navigation}/></Provider>);
        const image = getByTestId('login');
        fireEvent.press(image);
        expect(actions.login).toHaveBeenCalled();
      });
      test('Then navigation.navigate signup is invoked', () => {
        const { getByTestId } = render(<Provider store={store}><Login navigation={navigation}/></Provider>);
        const image = getByTestId('signup');
        fireEvent.press(image);
        expect(navigation.navigate).toHaveBeenCalled();
      });
})

describe('When invoked a Login component', () => {
  const mockStore = ConfigureStore();
  const store = mockStore({
      user: {
            token: 'bjwsvkjbskvbs',
            _id: '2',
            user : {
                alias: 'Anna',
                email: 'annita',
                password: 'annaaaza',
                avatar: 'img.com'
            }
        }
  });
  let navigation;
  beforeEach(() => {
      jest.spyOn(actions, 'logout').mockReturnValueOnce({ type: '' });
      jest.spyOn(actions, 'deleteAccount').mockReturnValueOnce({ type: '' });
      navigation = {
          navigate: jest.fn()
      };
    });

  test('Should render the login component with inputs', () => {
      const login = render(<Provider store={store}><Login navigation={navigation} /></Provider>)
      expect(login).toMatchSnapshot();
  })
    test('Then logout function is invoked', () => {
      const { getByTestId } = render(<Provider store={store}><Login navigation={navigation}/></Provider>);
      const image = getByTestId('logoutText');
      fireEvent.press(image);
      expect(actions.logout).toHaveBeenCalled();
    });
    test('Then navigation.navigate signup is invoked', () => {
      const { getByTestId } = render(<Provider store={store}><Login navigation={navigation}/></Provider>);
      const image = getByTestId('deleteAccountButton');
      fireEvent.press(image);
      expect(navigation.navigate).toHaveBeenCalled();
    });
})