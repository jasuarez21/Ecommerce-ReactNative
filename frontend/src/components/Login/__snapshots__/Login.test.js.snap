// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When invoked a Login component Should render the login component with inputs 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": 800,
      "width": 500,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://i.postimg.cc/htkHJcp0/fondo-INICIAL.jpg",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": 800,
          "width": 500,
        },
        undefined,
      ]
    }
  />
  <Image
    source={
      Object {
        "uri": "https://i.postimg.cc/FF2Y5GJd/logo-small.png",
      }
    }
    style={
      Object {
        "height": 42,
        "marginBottom": 30,
        "marginLeft": 50,
        "marginTop": 80,
        "width": 300,
      }
    }
  />
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "white",
        "fontSize": 16,
        "marginBottom": 30,
        "marginRight": 90,
        "marginTop": 50,
      }
    }
  >
    Rellena los campos para loguearte en nuestra pagina
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    defaultValue=""
    onChangeText={[Function]}
    placeholder="email"
    rejectResponderTermination={true}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "white",
        "borderRadius": 5,
        "marginBottom": 10,
        "marginRight": 70,
        "paddingLeft": 20,
        "width": 300,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    defaultValue=""
    onChangeText={[Function]}
    placeholder="password"
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "white",
        "borderRadius": 5,
        "marginBottom": 10,
        "marginRight": 70,
        "paddingLeft": 20,
        "width": 300,
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Image
      source={
        Object {
          "uri": "https://i.postimg.cc/6qnQSnBG/icons8-ok-26-1.png",
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "height": 35,
          "marginRight": 90,
          "marginTop": 5,
          "width": 35,
        }
      }
      testID="login"
    />
  </View>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "white",
        "fontSize": 16,
        "marginBottom": 30,
        "marginRight": 90,
        "marginTop": 50,
      }
    }
  >
    Todav√≠a no estas registrado?
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignContent": "center",
        "backgroundColor": "#FFB800",
        "borderRadius": 5,
        "display": "flex",
        "height": 35,
        "marginLeft": 185,
        "width": 40,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://i.postimg.cc/0jfF6Sbg/icons8-add-user-male-64.png",
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "height": 30,
          "marginTop": 1,
          "width": 30,
        }
      }
      testID="signup"
    />
  </View>
</View>
`;

exports[`When invoked a Login component Should render the login component with inputs 2`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": 800,
      "width": 500,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://i.postimg.cc/htkHJcp0/fondo-INICIAL.jpg",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": 800,
          "width": 500,
        },
        undefined,
      ]
    }
  />
  <Image
    source={
      Object {
        "uri": "https://i.postimg.cc/FF2Y5GJd/logo-small.png",
      }
    }
    style={
      Object {
        "height": 42,
        "marginBottom": 30,
        "marginLeft": 50,
        "marginTop": 80,
        "width": 300,
      }
    }
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text
      style={
        Object {
          "backgroundColor": "#FFB800",
          "borderRadius": 5,
          "color": "black",
          "marginLeft": 340,
          "marginTop": 20,
          "textAlign": "center",
          "width": 60,
        }
      }
      testID="logoutText"
    >
      SALIR
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "borderRadius": 37.5,
        "height": 75,
        "marginLeft": 180,
        "width": 75,
      }
    }
    theme={
      Object {
        "colors": Object {
          "black": "#242424",
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "grey": "rgba(0, 0, 0, 0.54)",
              "primary": "#2196f3",
              "searchBg": "#dcdce1",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "default": Object {
              "error": "#ff3b30",
              "grey": "#7d7d7d",
              "primary": "#007aff",
              "searchBg": "#dcdce1",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
            "ios": Object {
              "error": "#ff3b30",
              "grey": "#7d7d7d",
              "primary": "#007aff",
              "searchBg": "#dcdce1",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
            "web": Object {
              "error": "#ff190c",
              "grey": "#393e42",
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#ca71eb",
          "success": "#52c41a",
          "warning": "#faad14",
          "white": "#ffffff",
        },
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 37.5,
          "flex": 1,
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <Image
        borderRadius={37.5}
        onLoad={[Function]}
        source={
          Object {
            "uri": "img.com",
          }
        }
        style={
          Object {
            "bottom": 0,
            "flex": 1,
            "height": undefined,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": undefined,
          }
        }
        testID="RNE__Image"
        theme={
          Object {
            "colors": Object {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "searchBg": "#dcdce1",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "default": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": Object {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
        transition={true}
        transitionDuration={360}
      />
      <View
        accessibilityElementsHidden={true}
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#bdbdbd",
              "flex": 1,
              "height": undefined,
              "justifyContent": "center",
              "width": undefined,
            }
          }
          testID="RNE__Image__placeholder"
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "height": undefined,
            "width": undefined,
          }
        }
        testID="RNE__Image__children__container"
      />
    </View>
  </View>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "borderBottomColor": "#FFB800",
        "borderBottomWidth": 1,
        "color": "white",
        "fontSize": 18,
        "marginRight": 70,
        "marginTop": 30,
        "width": 300,
      }
    }
  >
    PERFIL
  </Text>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 18,
        "marginLeft": 70,
        "marginTop": 30,
      }
    }
  >
    Alias: 
    Anna
  </Text>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 18,
        "marginLeft": 70,
        "marginTop": 30,
      }
    }
  >
    Email: 
    annita
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text
      style={
        Object {
          "backgroundColor": "#FFB800",
          "borderRadius": 5,
          "color": "black",
          "marginLeft": 70,
          "marginTop": 40,
          "textAlign": "center",
          "width": 90,
        }
      }
      testID="deleteAccountButton"
    >
      Borrar Cuenta
    </Text>
  </View>
</View>
`;
